pkgs/native_assets_cli/lib/native_assets_cli.dart

```diff
@@ -6,13 +6,20 @@
 /// native assets CLI, e.g. a `build.dart` script.
 library native_assets_cli;
 
-export 'src/api/asset.dart';
-export 'src/api/build_config.dart';
-export 'src/api/build_mode.dart';
-export 'src/api/build_output.dart';
-export 'src/api/dependencies.dart';
-export 'src/api/ios_sdk.dart';
-export 'src/api/link_mode.dart';
-export 'src/api/link_mode_preference.dart';
-export 'src/api/metadata.dart';
-export 'src/api/target.dart';
+export 'src/api/architecture.dart' show Architecture;
+export 'src/api/asset.dart'
+    show
+        Asset,
+        BundledDylib,
+        CCodeAsset,
+        LookupInExecutable,
+        LookupInProcess,
+        SystemDylib;
+export 'src/api/build.dart';
```

Comment:
Consider avoiding  one-class-per-file, have one for `BuildConfig` and one for `BuildConfig.native`.

Having so many files makes it just hard to find things, forces one to navigate around all the time, ... I don't see the benefit.